### PR TITLE
Tamago ci

### PR DESCRIPTION
**Pull request by rminnich**

### CODE REVIEW

File: .circleci/images/test-image-tamago/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+RUN apt-get dist-upgrade -y
+
+# create user "builder" with sudo privileges
+ARG GID
+ARG UID
+ARG USER=builder
+RUN groupadd --gid ${GID} $USER
+RUN useradd --uid ${UID} --gid $USER --shell /bin/bash --home-dir /home/$USER --create-home $USER
+RUN apt-get install -y sudo
+RUN echo "builder ALL=(ALL) NOPASSWD: ALL" | tee -a /etc/sudoers
```
Comment (abrender) [inline .circleci/images/test-image-tamago/Dockerfile:13]: should this be `$USER` instead of `builder` ?

Comment (rminnich) [reply .circleci/images/test-image-tamago/Dockerfile:13]: I got kind of confused here, but IIUC builder is the in-container uid and UID and GID are the uid outside the container.

Comment (rminnich) [reply .circleci/images/test-image-tamago/Dockerfile:13]: it seems to be working ...

**Comment by rminnich** [comment]
ok, it works. 
andrea, we can get this in, then start expanding the set of things we build to make sure we keep tamago working

note I'm building for riscv64 and arm. Further, I'm happy to put in cmds that only run on tamago -- we have several commands that only run on Plan 9.

File: .circleci/images/test-image-tamago/Dockerfile (hunk 1 above)
Comment (rminnich) [reply .circleci/images/test-image-tamago/Dockerfile:13]: I tested it interactively and it seems fine.

File: .circleci/images/test-image-tamago/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                       u-boot-tools vim wget
+
+# install tamago-go
+ENV TAMAGO_VERSION="1.19.3"
+ENV TAMAGO_CHECKSUM="b614e5266e0933c67f00d77987d49e7fd290704c86482f97c8bb2c2e26e46e9f"
+RUN wget -O tamago-go.tgz https://github.com/usbarmory/tamago-go/releases/download/tamago-go${TAMAGO_VERSION}/tamago-go${TAMAGO_VERSION}.linux-amd64.tar.gz
+RUN echo "${TAMAGO_CHECKSUM} tamago-go.tgz" | sha256sum --strict --check -
+RUN tar -C / -xzf tamago-go.tgz && rm tamago-go.tgz
+
+# install minisign
```
Comment (abarisani) [inline .circleci/images/test-image-tamago/Dockerfile:29]: Minisign is actually not needed just for testing tamago builds, you can safely remove it.

File: .circleci/images/test-image-tamago/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ENV MINISIGN_VERSION="0.10"
+ENV MINISIGN_CHECKSUM="9fe40c2bd899a91f6b62a6ff3d469ece670f155307df50c2482ddd31337ab6da"
+RUN wget -O minisign.tgz https://github.com/jedisct1/minisign/archive/refs/tags/${MINISIGN_VERSION}.tar.gz
+RUN echo "${MINISIGN_CHECKSUM} minisign.tgz" | sha256sum --strict --check -
+RUN tar -xzf minisign.tgz
+RUN cd minisign-${MINISIGN_VERSION}; mkdir build; cd build; cmake ..; make; make install
+RUN rm -r minisign*
+
+# install habtool
+ENV HABTOOL_VERSION="v2022.05.25"
```
Comment (abarisani) [inline .circleci/images/test-image-tamago/Dockerfile:39]: The habtool is only needed to test secure boot signed binaries, this also has nothing to do with TamaGo and can be safely removed.

File: .circleci/images/test-image-tamago/Dockerfile (hunk 2 above)
Comment (rminnich) [reply .circleci/images/test-image-tamago/Dockerfile:29]: done

**Comment by rminnich** [comment]
we are not building the armory boot to test u-root usage in that component.

File: .circleci/config.yml [hunk 4]
```diff
@@ -85,6 +87,39 @@ jobs:
...
+    environment:
+      - UROOT_SOURCE: /home/circleci/go/src/github.com/u-root/u-root
+      - CGO_ENABLED: 0
+    steps:
+      - checkout
+      - run:
+          name: Test u-root tamago build
+          command: GOOS=tamago GOARCH=arm GOARM=7 go build ./pkg/dt/...
+
+  test-build-tamago-riscv64:
```
Comment (abarisani) [inline .circleci/config.yml:103]: armory-boot is not supported under riscv64, so this test doesn't make much sense to me.

Comment (rminnich) [reply .circleci/config.yml:103]: move it to arm I guess? This is mainly a did we break the build but your'r'e right :-)

**Review by abarisani — APPROVED** [review]

**Review by abarisani — APPROVED** [review]

**Pull request merged**